### PR TITLE
Layer cleanups (part 1)

### PR DESCRIPTION
**Pull request by rossburton**
Some small layer cleanups to start with...

### CODE REVIEW

File: recipes-graphics/mesa/mesa_%.bbappend [hunk 1]
```diff
@@ -1,6 +1,3 @@
-LDFLAGS:append:toolchain-clang = " -latomic -lm"
-DEPENDS:append:toolchain-clang = " libatomic-ops"
```
Comment (kraj) [inline recipes-graphics/mesa/mesa_%.bbappend:4]: I think the real test is mips and riscv32 or ppc32 where 64bit builtin atomics are missing

File: recipes-graphics/mesa/mesa_%.bbappend [hunk 2]
```diff
@@ -1,6 +1,3 @@
-EXTRA_OEMASON:append:toolchain-clang:x86 = " -Dasm=false"
-EXTRA_OEMASON:append:toolchain-clang:x86-64 = " -Dasm=false"
```
Comment (kraj) [inline recipes-graphics/mesa/mesa_%.bbappend:7]: hmmm interesting find.

File: recipes-graphics/mesa/mesa_%.bbappend (hunk 1 above)
Comment (rossburton) [reply recipes-graphics/mesa/mesa_%.bbappend:4]: `MACHINE=qemumips bitbake mesa-demos` just build successfully. Is that enough?

Comment (kraj) [reply recipes-graphics/mesa/mesa_%.bbappend:4]: seems good thanks

**Pull request merged**